File: problems/1310/solution.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        val, dp, ans = 0, [0] * len(arr), []
        
        for i in range(len(arr)):
            val ^= arr[i]
            dp[i] = val
        
        for li, ri in queries:
            if not li:
                ans.append(dp[ri])
            else:
                ans.append(dp[ri] ^ dp[li - 1])
        
        return ans

File: problems/1310/solution.go
package main
func xorQueries(arr []int, queries [][]int) []int {
    ans := make([]int, len(queries))
    
    for i, query := range queries {
        for j := query[0]; j <= query[1]; j++ {
            ans[i] ^= arr[j]
        }
    }
    
    return ans
}